spritz/spritz-android/src/main/java/com/novoda/spritz/Spritz.java

```diff
@@ -0,0 +1,257 @@
+                .setDuration(currentPage.autoPlayDuration());
+        autoPlayAnimation.addUpdateListener(defaultAnimatorUpdateListener());
+
+        return Arrays.<Animator>asList(finishSwipeAnimation, autoPlayAnimation);
+    }
+
+    private float getSwipeEndForPreviousPositionOrZero(int position) {
+        float swipeEndProgress = 0;
+        if (position > 0) {
+            swipeEndProgress = getSwipeEndProgressForPosition(position - 1);
+        }
+        return swipeEndProgress;
+    }
+
+    private TimeInterpolator getSwipeForwardInterpolatorFor(SpritzPageWithOffset currentPage) {
+        TimeInterpolator pageSwipeForwardInterpolator = currentPage.swipeForwardInterpolator();
+        if (pageSwipeForwardInterpolator != null) {
+            return pageSwipeForwardInterpolator;
+        }
+        return this.defaultSwipeForwardInterpolator;
+    }
+
+    private List<Animator> swipeBackwards(int position) {
+        SpritzPageWithOffset currentPage = spritzPagesWithOffset.get(position);
+        float currentProgress = lottieAnimationView.getProgress();
+        float autoPlayEndProgress = getAutoPlayEndProgressForPosition(position);
+
+        ValueAnimator finishSwipeAnimation = ValueAnimator
+                .ofFloat(currentProgress, autoPlayEndProgress)
+                .setDuration(defaultSwipeAnimationDuration);
+        finishSwipeAnimation.setInterpolator(getSwipeBackwardsInterpolatorFor(currentPage));
+        finishSwipeAnimation.addUpdateListener(defaultAnimatorUpdateListener());
+
+        return Collections.<Animator>singletonList(finishSwipeAnimation);
+    }
+
+    private float getAutoPlayEndProgressForPosition(int position) {
+        return ((float) spritzPagesWithOffset.get(position).autoPlayEnd()) / totalAnimationDuration;
+    }
+
+    private TimeInterpolator getSwipeBackwardsInterpolatorFor(SpritzPageWithOffset currentPage) {
+        TimeInterpolator pageSwipeBackwardsInterpolator = currentPage.swipeBackwardsInterpolator();
+        if (pageSwipeBackwardsInterpolator != null) {
+            return pageSwipeBackwardsInterpolator;
+        }
+        return this.defaultSwipeBackwardsInterpolator;
+    }
+
+    private ValueAnimator.AnimatorUpdateListener defaultAnimatorUpdateListener() {
+        return new ValueAnimator.AnimatorUpdateListener() {
+            @Override
+            public void onAnimationUpdate(ValueAnimator animation) {
+                lottieAnimationView.setProgress((Float) animation.getAnimatedValue());
+            }
+        };
+    }
+
+    public static class Builder {
+
+        private static final long DEFAULT_SWIPE_ANIMATION_DURATION = TimeUnit.MILLISECONDS.toMillis(500);
+
+        private final LottieAnimationView lottieAnimationView;
+
+        private long defaultSwipeAnimationDuration = DEFAULT_SWIPE_ANIMATION_DURATION;
+        private TimeInterpolator defaultSwipeForwardInterpolator = new LinearInterpolator();
```

Comment:
These two should be both `LinearOutSlowIn` by default (unless I have misunderstood and this is needed to decouple the animation from being 1:1 with the viewpager scroll events, in which case linear is OK as a default and I don't see many reasons to change it)

Reply:
The idea is that we don't know what the user wants, so we default to a linear animation in order to keep the original After Effects animation timing.